src/test/java/org/gridsuite/sensitivityanalysis/server/SensitivityAnalysisControllerTest.java

```diff
@@ -718,6 +726,61 @@ public void runAndSaveTest() throws Exception {
+                        .header(HEADER_USER_ID, "testUserId")
+                        .content(SENSITIVITY_INPUT_1))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+        assertEquals(RESULT_UUID, mapper.readValue(result.getResponse().getContentAsString(), UUID.class));
+
+        Message<byte[]> resultMessage = output.receive(TIMEOUT, "sensitivityanalysis.result");
+        assertEquals(RESULT_UUID.toString(), resultMessage.getHeaders().get("resultUuid"));
+        assertEquals("me", resultMessage.getHeaders().get("receiver"));
+
+        // check that the results is deterministic while the sort by contingency id is not (there are 2 results with the same id)
+        //so the results should be sorted by
+        ResultsSelector selectorNK = ResultsSelector.builder()
+                .functionType(SensitivityFunctionType.BRANCH_ACTIVE_POWER_1)
+                .tabSelection(ResultTab.N_K)
+                .sortKeysWithWeightAndDirection(Map.of(
+                        SortKey.CONTINGENCY, 1))
+                .pageSize(10)
+                .pageNumber(0)
+                .build();
+
+        result = mockMvc.perform(get("/" + VERSION + "/results/{resultUuid}?selector={selector}", RESULT_UUID, mapper.writeValueAsString(selectorNK)))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andReturn();
+        String bodyText = result.getResponse().getContentAsString();
+
+        //contingency.id comparator
+        Comparator<ContingencyEmbeddable> comparatorByContingencyId = comparing(ContingencyEmbeddable::getContingencyId, Comparator.comparing(String::toString));
+        //sensitivityId comparator (the toString is needed because UUID comparator is not the same as the string one)
+        Comparator<SensitivityEntity> comparatorBySensiId = comparing(s -> s.getSensitivityId().toString());
+        //contingency.id and resultUuid (in that order) comparator
+        Comparator<SensitivityEntity> comparatorByContingencyIdAndSensiId = comparing(SensitivityEntity::getContingency, comparatorByContingencyId).thenComparing(comparatorBySensiId);
+
+        List<SensitivityOfTo> sortedSensitivityList = (List<SensitivityOfTo>) sensitivityRepository.findAll().stream().filter(s -> s.getContingency() != null)
+                .sorted(comparatorByContingencyIdAndSensiId).map(sensitivityEntity ->
+                        SensitivityOfTo.builder().funcId(sensitivityEntity.getFactor().getFunctionId())
+                                .varId(sensitivityEntity.getFactor().getVariableId())
+                                .varIsAFilter(sensitivityEntity.getFactor().isVariableSet())
+                                .value(sensitivityEntity.getValue())
+                .functionReference(sensitivityEntity.getFunctionReference())
+                .build()).toList();
```

Comment:
You can extract that

Reply:
not sure what do you want me to extract here ?

Reply:
I mean all these lines only deserve to produce the sortedSensitivityList, so it can't be extracted in another method to have less lines in the test and make it more readable